src/Features/Core/Portable/CodeFixes/Suppression/AbstractSuppressionCodeFixProvider.PragmaBatchFixHelpers.cs

```diff
@@ -0,0 +1,197 @@
+                }
+
+                return textChanges;
+            }
+
+            private static async Task UpdateDiagnosticSpansAsync(Document currentDocument, Document newDocument, ImmutableArray<Diagnostic> diagnostics, Dictionary<Diagnostic, TextSpan> currentDiagnosticSpans, CancellationToken cancellationToken)
+            {
+                // Update the diagnostics spans based on the text changes.
+                var textChanges = await newDocument.GetTextChangesAsync(currentDocument, cancellationToken).ConfigureAwait(false);
+                foreach (var textChange in textChanges)
+                {
+                    UpdateDiagnosticSpans(diagnostics, currentDiagnosticSpans, textChange);
+                }
+            }
+
+            private static void UpdateDiagnosticSpans(ImmutableArray<Diagnostic> diagnostics, Dictionary<Diagnostic, TextSpan> currentDiagnosticSpans, TextChange textChange)
+            {
+                var isAdd = textChange.Span.Length == 0;
+                Func<TextSpan, bool> isPriorSpan = span => span.End <= textChange.Span.Start;
+                Func<TextSpan, bool> isFollowingSpan = span => span.Start >= textChange.Span.End;
+                Func<TextSpan, bool> isEnclosingSpan = span => span.Contains(textChange.Span);
+
+                foreach (var diagnostic in diagnostics)
+                {
+                    TextSpan currentSpan;
+                    if (!currentDiagnosticSpans.TryGetValue(diagnostic, out currentSpan))
+                    {
+                        continue;
+                    }
+
+                    if (isPriorSpan(currentSpan))
+                    {
+                        // Prior span, needs no update.
+                        continue;
+                    }
+
+                    var delta = textChange.NewText.Length - textChange.Span.Length;
+                    if (delta != 0)
+                    {
+                        if (isFollowingSpan(currentSpan))
+                        {
+                            // Following span.
+                            var newStart = currentSpan.Start + delta;
+                            var newSpan = new TextSpan(newStart, currentSpan.Length);
+                            currentDiagnosticSpans[diagnostic] = newSpan;
+                        }
+                        else if (isEnclosingSpan(currentSpan))
+                        {
+                            // Enclosing span.
+                            var newLength = currentSpan.Length + delta;
+                            var newSpan = new TextSpan(currentSpan.Start, newLength);
+                            currentDiagnosticSpans[diagnostic] = newSpan;
+                        }
+                        else
+                        {
+                            // Overlapping span.
+                            // Drop conflicting diagnostics.
+                            currentDiagnosticSpans.Remove(diagnostic);
```

Comment:
This should be very unlikely as it happens only if diagnostic span is partially hitting the restore/disable keyword on pragma - toggle pragma disable/restore is the only edit operation by this code action, others are add or remove operations.